pkg/cli/longhorn.go

```diff
@@ -0,0 +1,243 @@
+package cli
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strconv"
+	"time"
+
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+
+	lhv1b1 "github.com/longhorn/longhorn-manager/k8s/pkg/apis/longhorn/v1beta1"
+)
+
+var (
+	scaleDownReplicasWaitTime = time.Minute
+)
+
+const (
+	longhornNamespace       = "longhorn-system"
+	overProvisioningSetting = "storage-over-provisioning-percentage"
+)
+
+func NewLonghornCmd(cli CLI) *cobra.Command {
+	return &cobra.Command{
+		Use:   "longhorn",
+		Short: "Perform operations on a longhorn installation within a kURL cluster",
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+			return cli.GetViper().BindPFlags(cmd.PersistentFlags())
+		},
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			return cli.GetViper().BindPFlags(cmd.Flags())
+		},
+	}
+}
+
+func NewLonghornPrepareForMigration(_ CLI) *cobra.Command {
+	return &cobra.Command{
+		Use:          "prepare-for-migration",
+		Short:        "Prepares Longhorn for migration to a different storage provisioner.",
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			log.Print("Preparing Longhorn deployment for migration.")
+			cli, err := client.New(config.GetConfigOrDie(), client.Options{})
+			if err != nil {
+				return fmt.Errorf("error creating client: %s", err)
+			}
+			lhv1b1.AddToScheme(cli.Scheme())
+
+			var nodes corev1.NodeList
+			if err := cli.List(cmd.Context(), &nodes); err != nil {
+				return fmt.Errorf("error listing kubernetes nodes: %s", err)
+			} else if len(nodes.Items) == 1 {
+				log.Print("Only one node found, scaling down the number of volume replicas to 1.")
+				if err := scaleDownReplicas(cmd.Context(), cli); err != nil {
+					return fmt.Errorf("error scaling down longhorn replicas: %s", err)
+				}
+			}
+
+			unhealthy, err := unhealthyVolumes(cmd.Context(), cli)
+			if err != nil {
+				return fmt.Errorf("error assessing unhealthy volumes: %s", err)
+			}
+
+			if len(unhealthy) > 0 {
+				log.Printf("The following Longhorn volumes are unhealthy:")
+				for _, vol := range unhealthy {
+					log.Printf(" - %s/%s", longhornNamespace, vol)
+				}
+				return fmt.Errorf("error preparing longhorn for migration: unhealthy volumes")
```

Comment:
```suggestion
				return fmt.Errorf("error preparing longhorn for migration: unhealthy volumes. Be aware that the Longhorn  replicas were scaled down to do this check. We might want to scale them up")
```

@ricardomaraschini something like ^